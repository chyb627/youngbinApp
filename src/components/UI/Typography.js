import React from 'react';
import { Text as RNText } from 'react-native';

// export class Typography extends React.Component {
//   render() {
//     return (
//       <RNText style={{ color: this.props.color, fontSize: this.props.fontSize }}>
//         {this.props.children}
//       </RNText>
//     );
//   }
// }

export const Typography = (props) => {
  return <RNText style={{ color: props.color, fontSize: props.fontSize }}>{props.children}</RNText>;
};
