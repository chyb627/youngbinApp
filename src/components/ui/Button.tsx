import React, { ReactElement } from 'react';
import { TouchableOpacity } from 'react-native';

const Button: React.FC<{
  onPressIn?: () => void;
  onPressOut?: () => void;
  onPress: () => void;
  hitSlop?: { left: number; right: number; top: number; bottom: number };
  backgroundColor?: string;
  paddingHorizontal?: number;
  paddingVertical?: number;
  children: ReactElement | ReactElement[];
}> = (props) => (
  <TouchableOpacity
    {...props}
    onPressIn={props.onPressIn}
    onPressOut={props.onPressOut}
    onPress={props.onPress}
    hitSlop={
      props.hitSlop ?? {
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
      }
    }
    style={{
      backgroundColor: props.backgroundColor,
      paddingHorizontal: props.paddingHorizontal,
      paddingVertical: props.paddingVertical,
    }}>
    {props.children}
  </TouchableOpacity>
);

export default Button;
