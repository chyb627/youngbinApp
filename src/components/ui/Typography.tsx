import React from 'react';
import { Text as RNText } from 'react-native';

type TypographyProps = {
  color?: string;
  fontSize?: number;
  fontWeight?: 'normal' | 'bold' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900';
  numberOfLines?: number;
  lineHeight?: number;
  textDecorationLine?: 'none' | 'underline' | 'line-through' | 'underline line-through' | undefined;
  children: React.ReactNode | string;
};

const Typography: React.FC<TypographyProps> = (props) => (
  <RNText
    style={{
      color: props.color ?? 'black',
      fontWeight: props.fontWeight ?? 'normal',
      lineHeight: props.lineHeight && props.lineHeight,
      fontSize: props.fontSize ?? 10,
      textDecorationLine: props.textDecorationLine ?? 'none',
    }}
    numberOfLines={props.numberOfLines}>
    {props.children}
  </RNText>
);

export default Typography;
