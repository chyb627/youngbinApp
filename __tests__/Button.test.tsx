import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import Button from '../src/components/ui/Button';
import Typography from '../src/components/ui/Typography';

test('Button Test', async () => {
  const onPressed = jest.fn();
  const expectedButtonName = 'TEST_BUTTON';

  render(
    <Button onPress={onPressed}>
      <Typography>{expectedButtonName}</Typography>
    </Button>,
  );

  expect(screen.toJSON()).toMatchSnapshot();

  fireEvent.press(screen.getByText(expectedButtonName));

  expect(onPressed).toHaveBeenCalled();
});
