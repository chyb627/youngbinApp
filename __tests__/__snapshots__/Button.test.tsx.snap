// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": undefined,
      "opacity": 1,
      "paddingHorizontal": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 10,
        "fontWeight": "normal",
        "lineHeight": undefined,
        "textDecorationLine": "none",
      }
    }
  >
    TEST_BUTTON
  </Text>
</View>
`;
